### PR TITLE
Adding doctest for `fill-mask` pipeline.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Follow up of https://github.com/huggingface/transformers/pull/20226

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/pipelines/fill_mask.py [hunk 1]
```diff
@@ -36,6 +36,22 @@ class FillMaskPipeline(Pipeline):
...
+    ```python
+    >>> from transformers import pipeline
+
+    >>> fill_masker = pipeline(model="bert-base-uncased")
+    >>> potential_words = fill_masker("This is a simple [MASK].")
+
+    >>> from transformers.testing_utils import nested_simplify
+
+    >>> nested_simplify(potential_words)  # The scores might vary slightly across pytorch/tensorflow versions.
+    [{'score': 0.042, 'token': 3291, 'token_str': 'problem', 'sequence': 'this is a simple problem.'}, {'score': 0.031, 'token': 3160, 'token_str': 'question', 'sequence': 'this is a simple question.'}, {'score': 0.03, 'token': 8522, 'token_str': 'equation', 'sequence': 'this is a simple equation.'}, {'score': 0.027, 'token': 2028, 'token_str': 'one', 'sequence': 'this is a simple one.'}, {'score': 0.024, 'token': 3627, 'token_str': 'rule', 'sequence': 'this is a simple rule.'}]
```
Comment (sgugger) [inline src/transformers/pipelines/fill_mask.py:50]: I think the doctest will still pas if you use new lines instead of spaces for the result (which would make it more readable).

**Review by sgugger — APPROVED** [review]
LGTM, just one comment!

File: src/transformers/pipelines/fill_mask.py (hunk 1 above)
Comment (Narsil) [reply src/transformers/pipelines/fill_mask.py:50]: I could not make it work here for some reason... The test would fail because spacing was different... ??!!

It might be because I'm not using `assert` anymore for output, and the doctest checks for newlines...
